app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-enhanced-ecommerce-tracker.js

```diff
@@ -0,0 +1,144 @@
+        this is handled elsewhere. */
+        if (!this.isNewPageLoad) {
+          var pageViewTracker = window.GOVUK.analyticsGA4.analyticsModules.PageViewTracker
+
+          if (pageViewTracker) {
+            pageViewTracker.init()
+          }
+        }
+
+        for (var i = 0; i < this.searchResultsBlocks.length; i++) {
+          this.trackSearchResults(this.searchResultsBlocks[i])
+
+          if (this.isNewPageLoad) {
+            this.searchResultsBlocks[i].addEventListener('click', this.handleClick.bind(this))
+          }
+        }
+      }
+    },
+
+    trackSearchResults: function (searchResultsBlock) {
+      var schema = this.populateEcommerceSchema(searchResultsBlock)
+
+      this.clearPreviousEcommerceObject()
+      window.dataLayer.push(schema)
+    },
+
+    handleClick: function (event) {
+      var searchResultsBlock = event.target.closest('[data-ga4-ecommerce]')
+      var isSearchResult = event.target.getAttribute('data-ecommerce-path')
+
+      if (isSearchResult) {
+        var searchResult = event.target
+        var schema = this.populateEcommerceSchema(searchResultsBlock, true, searchResult)
+
+        this.clearPreviousEcommerceObject()
+        window.dataLayer.push(schema)
+      }
+    },
+
+    populateEcommerceSchema: function (searchResultsBlock, searchResultClicked = false, searchResult = null) {
+      // Limiting to 100 characters to avoid noise from extra long search queries and to stop the size of the payload going over 8k limit.
+      var searchQuery = this.PIIRemover.stripPII(searchResultsBlock.getAttribute('data-search-query')).substring(0, 100).toLowerCase()
+      var variant = searchResultsBlock.getAttribute('data-ecommerce-variant') || undefined
+      var ecommerceRows = searchResultsBlock.querySelectorAll('[data-ecommerce-row]')
+      var listTitle = searchResultsBlock.getAttribute('data-list-title') || this.DEFAULT_LIST_TITLE
+      var startPosition = parseInt(searchResultsBlock.getAttribute('data-ecommerce-start-index'), 10)
+
+      var ecommerceObject = {
+        event: 'search_results',
+        search_results: {
+          event_name: searchResultClicked && searchResult ? 'select_item' : 'view_item_list',
+          term: searchQuery,
+          sort: variant,
+          results: this.getResultsCount(searchResultsBlock),
+          ecommerce: {
+            items: []
+          }
+        }
+      }
+
+      // Populate items array
+      if (searchResultClicked && searchResult) {
+        ecommerceObject.search_results.ecommerce.items.push({
+          item_id: searchResult.getAttribute('data-ecommerce-path'),
+          item_name: searchResult.textContent,
+          item_list_name: listTitle,
+          index: this.getIndex(searchResult, startPosition)
+        })
+
+        GOVUK.analyticsGA4.analyticsModules.Ga4LinkTracker.internalLinksDomain = 'www.gov.uk/'
+        GOVUK.analyticsGA4.analyticsModules.Ga4LinkTracker.internalLinksDomainWithoutWww = 'gov.uk/'
```

Comment:
Hopefully a quick one. Can you recap why these need to be set? And will it have any detrimental effect on the rest of the code once these two lines have run?

Reply:
I’ve just pushed up a change that removes these. Initially, we were having problems without setting it but I’ve just tested this locally and everything works fine without them. The only place where we’d need to keep them is in the tests as the link tracker isn’t initialised there.